### DISCUSSION TITLE
How can we upload our attack in this library?

### DISCUSSION THREAD

**Discussion by aliotopal** [Trusted-AI/adversarial-robustness-toolbox#2158] [Q&A]
We have been working on our own attack which is Evolutionary Algorithm based black box attack. 

Here, you can find the related paper and the code:
Paper: https://www.sciencedirect.com/science/article/pii/S1568494623004155  (open access)
code: https://github.com/aliotopal/EAbasedAttack

I've already adapted the code so it looks like evasion attacks in your library. 

Can we include our EAattack in your library for anyone to use?

**Comment by beat-buesser** [comment]
Hi @aliotopal Thank you very much for your interest in contributing to ART! I'm somehow not able to get to the paper without account, do you also have your paper published on Arxiv?

**Reply by aliotopal** [reply]
Hi,
No, we did not publish on Arxiv. But you should be able to open without an account. We paid to make it open access. I will have a look.

**Reply by aliotopal** [reply]
Can you please try this link for the paper: https://doi.org/10.1016/j.asoc.2023.110397 

I have also added the paper in GitHub: 
https://github.com/aliotopal/EAbasedAttack/blob/master/EAAttack_ASC_2023.pdf

**Reply by beat-buesser** [reply]
Hi @aliotopal Ok, this link works for me. Thank you for citing ART! Let me take a closer look a the paper. In the meantime you could open a draft pull request with your current code starting from branch `dev_1.15.0` and implement your attack following the other evasion attacks. Let me know when you have questions!

**Reply by aliotopal** [reply]
Hi @beat-buesser, I implemented our attack and added in dev_1.15.0 ( as far as I understood, I don't shine when it comes to GitHub ), can you please have a look if it is ok.
